server/src/main/java/umm3601/product/ProductController.java

```diff
@@ -126,10 +125,20 @@ public void groupProductsByCategory(Context ctx) {
                         Projections.computed("category", "$_id"),
                         Projections.include("count", "products"),
                         Projections.excludeId())),
-                Aggregates.sort(Sorts.ascending("_id"))),
+                Aggregates.sort(Sorts.ascending("category"))),
             CategorySortItem.class)
         .into(new ArrayList<>());
 
+    // For each product category, this sorts the list of products in that
+    // category by the product name.
+    // It would be nice to figure out how to have Mongo do this internal
+    // sorting for us, but this works for now.
+    output.forEach(categoryEntry -> {
+      categoryEntry.products.sort((first, second) -> {
```

Comment:
Are there other places where we might want to sort by product name? If so, do we want to pull this out into a named function so it can be re-used?